src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs

```diff
@@ -39,7 +39,15 @@ public async Task ExecuteAsync()
 			await Task.WhenAll(files.Select(file 
 				=> NavigationHelpers.OpenPath(file.path, _pageContext.ShellPage!)));
 
-			folders.ForEach(async folder => await NavigationHelpers.OpenPathInNewTab(folder.path, false));
+			var newTabsTasks = folders.Select(async folder =>
+			{
+				await NavigationHelpers.OpenPathInNewTab(folder.path, false);
```

Comment:
You should not put `await` here. What you want to do is returning the task then you make `await Task.WhenAll(newTabsTasks);`

Reply:
Done, you're 100% right, I removed `async void` from ForEach and forgot that `Select` also don't support `Task`.